test/go.mod

```diff
@@ -14,15 +14,15 @@ require (
 	gopkg.in/yaml.v2 v2.4.0
 	k8s.io/api v0.31.4
 	k8s.io/apiextensions-apiserver v0.31.4
-	k8s.io/apimachinery v0.31.4
+	k8s.io/apimachinery v0.32.1
```

Comment:
This makes k8s modules being out of sync.

We have done k8s bumps, especially if there is new controller-tools/runtime available as separate PR. That would bump the Go dependency as well. 

We could do that in separate PR, and then here we can let dependabot focus on bumping just kustomize on its own?

Reply:
I'm not sure, but we could try.

Reply:
We can't do that yet. CAPI v1.9.3 still uses utilversion from "k8s.io/apiserver/pkg/util/version". It prevents use from uplifting apiserver package to v0.32

Reply:
It seems to be a 1.10 item. apiserver/pkg/util/version was changed to "k8s.io/component-base/version" in main but not release-1.9 branch.

Reply:
OK. CAPI 1.10 will be out before we do our next minor, so we'll get to sort k8s deps by that time too.

Reply:
so, a bit confused , I also agree that we should keep our k8s deps in sync with CAPI, so why are we taking this in then? Cant we wait until CAPI does it?

Reply:
"in sync", I did not refer to being in sync with CAPI, but being in sync within the repo (not mixing v0.31 and v0.32 deps). CAPI will fix their k8s support by v1.10, allowing up to bump all k8s deps to v0.32.

Also, similar PR has been taken in in CAPM3 too with no issues.

Reply:
ok, but usually we do keep k8s minor bumps in sync with CAPI. Its in `test/` so I am ok to take this in.